src/phoneField/index.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { PureComponent } from 'react'
+import cc from 'classcat'
+import isEmpty from 'lodash.isempty'
+import isEqual from 'lodash.isequal'
+import SelectField from 'selectField'
+import TextField from 'textField'
+import countryTelData from 'country-telephone-data'
+import getCountryFlag from '_utils/countriecodes'
+import style from 'phoneField/style'
+
+interface PhoneFieldOnChangeParameters {
+  name: string
+  value: object
+}
+interface PhoneFieldProps  {
+  readonly id?: string
+  readonly name: string
+  readonly className?: string
+  readonly defaultPhoneCountryCodeValue?: string
+  readonly PhoneCountryCodeLabel?: string
+  readonly phoneNumberLabel?: string
+  readonly phoneNumberPlaceholder?: string
+  readonly defaultphoneNumberValue?: string
+  readonly countriesWhiteList?: string[]
+  readonly onChange?: (obj: PhoneFieldOnChangeParameters) => void
+}
+
+interface PhoneFieldState {
+  countriesDatas: object[]
+  countriesWhiteList: string[]
+  phoneNumber: string
+  phoneCountryCode: string
+  completePhoneNumber: string
+  // Make the type of [key: string]: string etc... assignable for setState()
+  [key: string]: string | string[] | object[] | number | boolean
+}
+
+const filterIso2 = (countriesList: string[]) => countryTelData.allCountries.filter(
+  country => countriesList.includes(country.iso2)
+)
+
+const mapCountriePhoneDatas = (param: allCountriesPhoneDatas[]):
+  mappedCountryPhoneDatas[] => param.map(countryDatas => ({
+    value: `+${countryDatas.dialCode}`,
+    label: `${getCountryFlag(countryDatas.iso2)} +(${countryDatas.dialCode})`,
+    ariaLabel: countryDatas.name,
+  })
+)
+
+const allCountriesPhoneDatas = mapCountriePhoneDatas(countryTelData.allCountries)
```

Comment:
Typo: country/countries

Reply:
`countryTelData.allCountries` is from the `country-telephone-data` library.

Reply:
I was talking about this method: `mapCountriePhoneDatas`

Reply:
Done